fsharp-backend/src/LibBackend/Serialize.fs

```diff
@@ -64,6 +64,62 @@ let isLatestOpRequest
 // --------------------------------------------------------
 module OT = LibExecution.OCamlTypes
 
+let deserializeOCamlSerializedToplevel
+  (tlid : tlid)
+  (typ : string)
+  (ocamlSerialized : byte [])
+  (pos : Option<string>)
+  : Task<PT.Toplevel> =
+  let deserializePos (pos : Option<string>) : pos =
+    try
+      pos
+      |> Option.map Json.OCamlCompatible.deserialize<pos>
+      |> Option.unwrap { x = 0; y = 0 }
+    with
+    | e -> { x = 0; y = 0 }
+  match typ with
+  | "db" ->
+    task {
```

Comment:
Does this compile/work if you put the `match` inside the `task {}`, rather than have a `task` in each branch? If so, that'd read a bit nicer I think

Reply:
It didn't work when I did that initially, even though I expected it would. Since this code isn't going to survive too long, i didn't spend much time investigating.